### PR TITLE
feat(ComponentPortlet & ImportCDX): Validate VCS URL

### LINKED ISSUE #2377: VCS Validation of a component and sanitization of github URL's. Change CDX Importer logic accordingly.
[//]: # (Copyright Siemens AG, 2021. Part of the SW360 Portal Project)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

### Description

- User can put any URL in the VCS field during component creation/editing such as "ettp://a23". Only URLS starting with "https://" should be allowed.
- Once the request is submitted, a check should be carried out in the backend for VCS of github.com domain (later more domains like bitbucket, gitlab, google code can be taken up).- It should follow the format: https://github.com/supplier/name. URLs like these: `git+https://github.com/angular/angular.git#packages/animations` `git+ssh://git@github.com/constellation/escodegen` `git://github.com/lgalfaso/angular-dynamic-locale` should be sanitized before allowing in the DB to enable one VCS per component policy in SW360.
- Appropriate error to be thrown when the GitHub URL has missing parts like component name.
- This check should be introduced at the CDX Importer level as well (reuse the same sanitizeVCS function).

#### Steps followed and expected result

- This VCS Validation mechanism is essential for the SRC Upload service of releases to work correctly.

### PR DESCRIPTION
**Pull request by afsahsyeda**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

**Description:**

- The VCS URL if not empty, will be validated for a valid URL while creating and updating a component. If the VCS string contains "github.com", it will sanitized to remove the noise and conform to the repository URL format: https://github.com/{supplier}/{component name}.
- Packages with invalid VCS URLs in the SBOM will be created as orphan packages

Closes #2377 

### How To Test?

1. Creating/Updating a component via REST API and UI.
2. Creating a component via CDX Import.

_Relevant Screenshots:_
image

### CODE REVIEW

**Comment by rajaraajeshwari** [comment]
Hi @afsahsyeda
Could you please let us know when this PR will be merged and this feature will be available.

File: frontend/sw360-portlet/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -858,7 +858,7 @@ link.to.project=Link to Project
 link.to.projects=Link to Projects
 list.of.components.without.version.information=List of Components without version information
 list.of.invalid.packages.without.purl.or.name.or.version=List of invalid Packages without purl or name or version
-list.of.packages.without.vcs.information=List of Packages without VCS information
+list.of.packages.without.vcs.information=List of Packages with invalid or missing VCS information
```
Comment (afsahsyeda) [inline frontend/sw360-portlet/src/main/resources/content/Language.properties:861]: This a workaround until we migrate to React framework. Did not want to touch the jsp files involving liferay variables.

**Comment by akshitjoshii** [comment]
Testing this PR

**Comment by akshitjoshii** [comment]
Testing was successful. VCS validation and github vcs sanitization happening while importing SBOM and manual comp. creation as well.
vcs_validation1


**After importing :** 



vcs_validation2

Error msg for incomplete/wrong github vcs also working.

**Comment by FarooqAftab** [comment]
@akapti  @afsahsyeda vcs field is not  getting updated via rest api 
image

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Comment by GMishx** [comment]
@afsahsyeda please rebase the branch with latest main without the frontend changes.

**Comment by afsahsyeda** [comment]
@GMishx Please code review the PR again as I have made a few additional changes.
@akshitjoshii Please test the PR again as a lot of frontend files have been removed.

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Comment by akshitjoshii** [comment]
Test was successful. 


**SBOM containing contaminated VCS**

image

**Sanitized VCS after import :**

image

**Pull request merged**